### DISCUSSION TITLE
Pagination issue

### DISCUSSION THREAD

**Discussion by paur94** [KevinVandy/material-react-table#552] [Q&A] [answered]
Hi there!
I have an issue with table pagination. When I go forward 2 or more pages and change the column, which is dropDown, it goes to the first page. How can I fix that or retrieve the current page?
I used this function, but it rerenders my component many times:
onPaginationChange={(updater) => {
    setPagination((prev) =>
        updater instanceof Function ? updater(prev) : updater,
    );
    queueMicrotask(rerender);
}}

**Comment by KevinVandy** [comment, marked as answer]
It could be because the reference to your data is changing for some reason. Fix that or set autoResetPageIndex prop to false

**Comment by hamdanmalekji** [comment]
Thanks, @KevinVandy
